### PR TITLE
boards: set CONFIG_BOARD default based on devicetree

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
This is cool, but makes some problems we have worse.

@tejlmand and I have been talking about how it's a problem that CONFIG_BOARD in Kconfig doesn't have to match BOARD in CMake. You see this most often in Arm v8-M boards with trustzone, where `BOARD=foo_ns` and `BOARD=foo` both have `CONFIG_BOARD=foo`. That's a bug in our opinion; consistency between BOARD and CONFIG_BOARD should be enforced. This PR would make that harder to do.

**Comment by tejlmand** [comment]
Agree with @mbolivar-nordic.

The principle is cool, but makes inconsistency worse.
Another issue that will arise is the sysbuild intregration.

Already now it's ~hard~ impossible to reuse the Kconfig tree for boards and SoC, but ideally we would like to be able to source part of the tree into sysbuild.
Namely the generic parts containing names but potentially other things, this is not fully decided.
Reason for this, is one may be like to have images available based on board / SoC / hw.

For example enabling a netcore image, such as HCI RPMsg, when building on a nRF5340, but not other boards.
Same could be true for features like Trustzone.

See also: https://github.com/zephyrproject-rtos/zephyr/issues/43488

**Comment by galak** [comment]
> This is cool, but makes some problems we have worse.
> 
> @tejlmand and I have been talking about how it's a problem that CONFIG_BOARD in Kconfig doesn't have to match BOARD in CMake. You see this most often in Arm v8-M boards with trustzone, where `BOARD=foo_ns` and `BOARD=foo` both have `CONFIG_BOARD=foo`. That's a bug in our opinion; consistency between BOARD and CONFIG_BOARD should be enforced. This PR would make that harder to do.

@mbolivar-nordic @tejlmand so why don't we effectively add -DCONFIG_BOARD=<BOARD> to set this to match and remove the Kconfig.defconfig setting of CONFIG_BOARD?

File: dts/bindings/base/base.yaml [hunk 1]
```diff
@@ -59,6 +59,11 @@ properties:
         deprecated: true
         description: Human readable string describing the device (used as device_get_binding() argument)
 
+    model:
+        type: string
+        required: false
+        description: specifies the manufacturer’s model number of the device
```
Comment (erwango) [inline dts/bindings/base/base.yaml:65]: s/number/name

**Comment by mbolivar-nordic** [comment]
>  why don't we effectively add -DCONFIG_BOARD= to set this to match and remove the Kconfig.defconfig setting of CONFIG_BOARD?

I don't get what you mean, sorry.

**Comment by galak** [comment]
> I don't get what you mean, sorry.

I mean if CONFIG_BOARD is suppose to match the cmake -DBOARD, why don't we enforce/pass that through from Cmake to Kconfig.  Instead of having CONFIG_BOARD set in Kconfig files?

**Comment by tejlmand** [comment]
> > I don't get what you mean, sorry.
> 
> I mean if CONFIG_BOARD is suppose to match the cmake -DBOARD, why don't we enforce/pass that through from Cmake to Kconfig. Instead of having CONFIG_BOARD set in Kconfig files?

work has been taken up by @nordicjm on this, if curious, see this commit for details: https://github.com/zephyrproject-rtos/zephyr/pull/50154/commits/ab05d915496d40f0007a9f95f7aa15960ba2bb9a

**Comment by mbolivar-nordic** [comment]
> see this commit for details: [ab05d91](https://github.com/zephyrproject-rtos/zephyr/commit/ab05d915496d40f0007a9f95f7aa15960ba2bb9a)

This commit isn't helpful; do you have a branch pointer?

**Comment by galak** [comment]
> This commit isn't helpful; do you have a branch pointer?

I assume https://github.com/zephyrproject-rtos/zephyr/pull/50154 is related to what @tejlmand mentioned.

**Comment by tejlmand** [comment]
> This commit isn't helpful; do you have a branch pointer?

@mbolivar-nordic  github is playing tricks here.
Cause the link takes you directly to a commit, without PR information, but if you look at the real link in the message (click `edit` on that message), then this is the link it contains:
```
https://github.com/zephyrproject-rtos/zephyr/pull/50154/commits/ab05d915496d40f0007a9f95f7aa15960ba2bb9a
```
which includes the PR number 50154, which is why I posted the link like that.
The interesting PR is part of the commit link, so everyone should be able to locate both pieces of information 😄

**Pull request closed without merging**